cns/service/main.go

```diff
@@ -880,6 +884,43 @@ func main() {
 		}
 	}
 
+	if cnsconfig.EnableSwiftV2 && cnsconfig.EnableK8sDevicePlugin {
+		initialVnetNICCount := 0
+		initialIBNICCount := 0
+		// Create device plugin manager instance
+		pluginManager := deviceplugin.NewPluginManager(z)
+		pluginManager.AddPlugin(mtv1alpha1.DeviceTypeVnetNIC, initialVnetNICCount)
+		pluginManager.AddPlugin(mtv1alpha1.DeviceTypeInfiniBandNIC, initialIBNICCount)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// Start device plugin manager in a separate goroutine
+		go func() {
+			for {
+				select {
+				case <-ctx.Done():
+					z.Info("Context canceled, stopping plugin manager")
+					return
+				default:
+					if pluginErr := pluginManager.Run(ctx); pluginErr != nil {
+						z.Error("plugin manager exited with error", zap.Error(pluginErr))
+						time.Sleep(defaultDevicePluginRetryInterval)
+					} else {
+						return
+					}
```

Comment:
This construction is what makes me think the `for` should, at least, be more tightly placed around the `pluginManager`'s `Run()`.

Reply:
We want to try running plugin only after a certain interval, hence the "for select case" structure around the pluginManager's Run()